frontend/src/app/services.ts

```diff
@@ -41,6 +42,25 @@ export class Config {
 // Helper to return either the passed value prefixed with `/` or an empty string
 const idOrBlank = (id: string | undefined) => (id ? `/${id}` : "");
 
+export const downloadApartmentMaximumPricePDF = (apartment: IApartmentDetails) => {
+    if (!apartment.prices.maximum_prices.confirmed) {
+        hitasToast("Enimmäishintalaskelmaa ei ole olemassa", "error");
+        return;
+    }
+
+    const url = `${Config.api_url}/housing-companies/${apartment.links.housing_company.id}/apartments/${apartment.id}/maximum-prices/${apartment.prices.maximum_prices.confirmed.id}/download`;
+    const init = {headers: new Headers({Authorization: "Bearer " + Config.token})};
+    fetch(url, init).then((response) => {
+        response.blob().then((blob) => {
+            const filename = `Enimmäishintalaskelma ${apartment.address.street_address} ${apartment.address.stair} ${apartment.address.apartment_number}`;
```

Comment:
how about returning the name as part of the server response (with `Content-Disposition`)